src/Library/demos/Box/main.blp

```diff
@@ -0,0 +1,202 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.Clamp {
+  maximum-size: 1024;
+
+  Box {
+    orientation: vertical;
+
+    Label {
+      label: "Box";
+      margin-top: 12;
+      margin-bottom: 12;
+      styles [
+        "title-1",
+      ]
+    }
+
+    Label {
+      label: _("A widget that arranges its child widgets into a single row or column");
+    }
+
+    Box{
+      halign: center;
+      valign: end;
+      margin-bottom: 12;
+
+      LinkButton{
+        label: "Tutorial";
+        uri: "https://developer.gnome.org/documentation/tutorials/beginners/components/box.html";
+      }
+
+      LinkButton{
+        label: "API Reference";
+        uri: "https://docs.gtk.org/gtk4/class.Box.html";
+      }
+    }
+
+    Box {
+      orientation: vertical;
+
+      Box {
+        hexpand: true;
+        orientation: vertical;
+        spacing: 12;
+        margin-bottom:24;
+
+        Box {
+          halign: center;
+          spacing: 18;
+
+          Button button_append{
+            label: _("Append Item");
+            styles [
+              "pill",
+            ]
+          }
+
+          Button button_prepend{
+            label: _("Prepend Item");
+            styles [
+              "pill",
+            ]
+          }
+
+          Button button_remove{
+            label: _("Remove Item");
+            styles [
+              "pill",
+            ]
+          }
+        }
+      }
+
+        Box {
+          halign:center;
+          spacing:18;
+
+          Box{
+            Label {
+              label: _("orientation");
+            }
+
+            Box {
+              margin-start: 6;
+              homogeneous: true;
+              halign: center;
+
+              ToggleButton toggle_orientation_horizontal {
+                label: _("horizontal");
+                active: true;
+              }
+              ToggleButton toggle_orientation_vertical {
+                label: _("vertical");
+                group: toggle_orientation_horizontal;
+              }
+            }
+          }
+
+          CheckButton highlight{
+            label: _("Highlight Box");
+          }
+      }
+
+      Grid {
```

Comment:
This is definitely the use for a grid, however I recommend you simplify the layout a bit.

I think this would best laid out so that (col, row):

* nothing at `0, 0`
* "halign" box is at `1, 0`
* "valign" box at `0, 1`
* `ScrolledWindow` at `1,1`

And just remove the `column-span`/`row-span` so they default to `1`. You can then move the `margin-end` from `ScrolledWindow` to `Grid` as necessary.

Reply:
I actually chose that layout because I was trying to center all the options. But because of it, like you said I had to put a nasty ``margin-end: 114;`` on the ``ScrolledWindow``

![image](https://user-images.githubusercontent.com/58233418/224610733-8e4cb2fe-7494-4bc7-9245-491bb36b3733.png)

With this layout
![image](https://user-images.githubusercontent.com/58233418/224610905-2cd164c1-6e46-44f0-8d19-c160efb24c5f.png)

I'm not sure how to tackle this otherwise, unless centering that is not very important which is fine 👍🏾

Reply:
Okay this works 👍🏾